TYPO3.Flow/Classes/TYPO3/Flow/Http/Request.php

```diff
@@ -373,6 +382,156 @@ public function getContent($asResource = false)
+     *
+     * This method obviates the need for a hasAttribute() method, as it allows
+     * specifying a default value to return if the attribute is not found.
+     *
+     * @see getAttributes()
+     * @param string $name The attribute name.
+     * @param mixed $default Default value to return if the attribute does not exist.
+     * @return mixed
+     */
+    public function getAttribute($name, $default = null)
+    {
+        if (isset($this->attributes[$name])) {
+            return $this->attributes[$name];
+        }
+        return $default;
+    }
+
+    /**
+     * PSR-7
+     *
+     * Return an instance with the specified derived request attribute.
+     *
+     * This method allows setting a single derived request attribute as
+     * described in getAttributes().
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that has the
+     * updated attribute.
+     *
+     * @see getAttributes()
+     * @param string $name The attribute name.
+     * @param mixed $value The value of the attribute.
+     * @return self
+     */
+    public function withAttribute($name, $value)
+    {
+        $request = clone $this;
+        $request->setAttribute($name, $value);
+        return $request;
+    }
+
+    /**
+     * @param string $name The attribute name.
+     * @param mixed $value The value of the attribute.
+     */
+    protected function setAttribute($name, $value)
+    {
+        $this->attributes[$name] = $value;
+    }
+
+    /**
+     * PSR-7
+     *
+     * Return an instance that removes the specified derived request attribute.
+     *
+     * This method allows removing a single derived request attribute as
+     * described in getAttributes().
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that removes
+     * the attribute.
+     *
+     * @see getAttributes()
+     * @param string $name The attribute name.
+     * @return self
+     */
+    public function withoutAttribute($name)
+    {
+        $request = clone $this;
+        $request->unsetAttribute($name);
+        return $request;
+    }
+
+    /**
+     * @param string $name The attribute name.
+     */
+    protected function unsetAttribute($name)
+    {
+        unset($this->attributes[$name]);
+    }
+
+    /**
+     * @param string $clientIpAddress
+     * @return Request
+     */
+    public function withClientIpAddress($clientIpAddress)
+    {
+        return $this->withAttribute('clientIpAddress', $clientIpAddress);
+    }
+
+    /**
+     * Check if this request is coming from a trusted proxy.
+     *
+     * @return boolean
+     */
+    public function isFromTrustedProxy()
+    {
+        return $this->getAttribute('fromTrustedProxy', false);
+    }
+
+    /**
+     * @return Request
+     */
+    public function fromTrustedProxy()
```

Comment:
Yes, guess it makes sense to just set the attribute or an HTTP header for this as @mficzel suggested.